featuretools/tests/primitive_tests/test_feature_serialization.py

```diff
@@ -94,3 +102,69 @@ def serialize_name_unchanged(original):
 
     for feature_type in feature_type_list:
         serialize_name_unchanged(feature_type)
+
+
+@mock_s3
+def test_serialize_features_mock_s3(es):
+    features_original = ft.dfs(target_entity='sessions', entityset=es, features_only=True)
+    s3 = boto3.resource('s3')
+    s3.create_bucket(Bucket=BUCKET_NAME, ACL='public-read-write')
+    url = "s3://{}/{}".format(BUCKET_NAME, WRITE_KEY_NAME)
+
+    ft.save_features(features_original, url)
+
+    bucket = s3.Bucket(BUCKET_NAME)
+    obj = list(bucket.objects.all())[0].key
+    s3.ObjectAcl(BUCKET_NAME, obj).put(ACL='public-read-write')
+
+    features_deserialized = ft.load_features(url)
+
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
+        assert feat_1.unique_name() == feat_2.unique_name()
+        assert feat_1.entityset == feat_2.entityset
+
+
+def test_deserialize_features_s3(es):
+    # TODO: Feature ordering is different in py3.5 vs 3.6+
+    features_original = sorted(ft.dfs(target_entity='sessions', entityset=es, features_only=True), key=lambda x: x.unique_name())
+    url = "s3://featuretools-static/test_feature_serialization_1.0.0"
+
+    features_deserialized = sorted(ft.load_features(url), key=lambda x: x.unique_name())
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
+        assert feat_1.unique_name() == feat_2.unique_name()
+        assert feat_1.entityset == feat_2.entityset
+
+
+def test_deserialize_features_url(es):
+    # TODO: Feature ordering is different in py3.5 vs 3.6+
+    features_original = sorted(ft.dfs(target_entity='sessions', entityset=es, features_only=True), key=lambda x: x.unique_name())
+    url = "https://featuretools-static.s3.amazonaws.com/test_feature_serialization_1.0.0"
+
+    features_deserialized = sorted(ft.load_features(url), key=lambda x: x.unique_name())
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
+        assert feat_1.unique_name() == feat_2.unique_name()
+        assert feat_1.entityset == feat_2.entityset
+
+
+def test_serialize_url(es):
+    features_original = ft.dfs(target_entity='sessions', entityset=es, features_only=True)
+    url = "https://featuretools-static.s3.amazonaws.com/test_feature_serialization_1.0.0"
+
+    error_text = "Writing to URLs is not supported"
+    with pytest.raises(ValueError, match=error_text):
+        ft.save_features(features_original, url)
+
+
+def tests_s3_profile_serialize(es):
+    features_original = ft.dfs(target_entity='sessions', entityset=es, features_only=True)
+    test_url = "s3://featuretools-static/test_feature_serialization_1.0.0"
+    error_text = "The config profile (.*) could not be found"
+    with pytest.raises(ProfileNotFound, match=error_text):
+        ft.save_features(features_original, test_url, profile_name="aws")
```

Comment:
This test can fail unexpectedly if the user running the tests has a profile named "aws"

Reply:
Should this test as a whole be changed? This was added to test if the AWS profiles are correctly passed on. As a fix I've made the profile name more random.